plugin/pulpcore/plugin/models/remote.py

```diff
@@ -34,28 +37,51 @@ def download_factory(self):
 
         Returns:
             DownloadFactory: The instantiated DownloaderFactory to be used by
-                get_downloader()
+                get_downloader().
         """
         try:
             return self._download_factory
         except AttributeError:
             self._download_factory = DownloaderFactory(self)
             return self._download_factory
 
-    def get_downloader(self, url, **kwargs):
+    def get_downloader(self, remote_artifact=None, url=None, **kwargs):
```

Comment:
Doesn't this break (some) existing uses of `get_downloader`? This MR should have a breaking tag.  Should `url` come first?

Reply:
Yes it does, but I forgot to set that. I added it just now. I actually put url second to make it super backwards-incompatible so that errors from this change won't pass silently. Thank you @gmbnomis